var/spack/repos/builtin/packages/elfutils/package.py

```diff
@@ -104,15 +106,11 @@ def configure_args(self):
         spec = self.spec
         args = []
 
-        if "+bzip2" in spec:
-            args.append("--with-bzlib=%s" % spec["bzip2"].prefix)
-        else:
-            args.append("--without-bzlib")
+        args.extend(self.with_or_without("bzip2", activation_value="prefix"))
+        args.extend(self.with_or_without("xz", activation_value="prefix"))
 
-        if "+xz" in spec:
-            args.append("--with-lzma=%s" % spec["xz"].prefix)
-        else:
-            args.append("--without-lzma")
+        with when("zstd"):  # zstd variant is available
```

Comment:
Interesting use of `with when`.  This is only the second package I've seen  use it for arguments.  Usually it's use to conditionally add variants, dependencies, and other directives.

I think you're missing the "+" for indicating you want to extend the args when the variant is enabled:

```suggestion
        with when("+zstd"):  # zstd variant is available
```

Reply:
In fact I wanted to only call `with_or_without` when variant `zstd` existed. However, this case is already handled by `with_or_without` according to documentation.